Terminal.Gui/Views/TextMaskField.cs

```diff
@@ -0,0 +1,535 @@
+﻿//
+// TextMaskField.cs: single-line text editor with Emacs keybindings
+//
+// Authors:
+//   Miguel de Icaza (miguel@gnome.org)
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NStack;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// TextField with Mask
+	/// </summary>
+	public class TextMaskField : View {
+
+		/// <summary>
+		/// Mask Type 
+		/// </summary>
+		public enum TextMaskType {
+			/// <summary>
+			/// Validates input with <see cref="MaskRunes"/>.
+			/// </summary>
+			Mask,
+			/// <summary>
+			/// Validetes input with a Regex Pattern.
+			/// </summary>
+			Regex
+		}
+
+		Regex regex;
+		List<Rune> text;
+		List<Rune> mask;
+		TextMaskType masktype;
+		bool error = false;
+
+		/// <summary>
+		/// When masktype == regex and text is right aligned, cursorPosition goes from 1 to x, leaving 0 for insert.
+		/// </summary>
+		int cursorPosition;
+
+		/// <summary>
+		/// Raised when the <see cref="Text"/> of the <see cref="TextMaskField"/> changes.
+		/// </summary>
+		public event Action<ustring> TextChanged;
+
+		/// <summary>
+		/// Raised when the validation of the <see cref="TextMaskField"/> occurs.
+		/// </summary>
+		public event Action<bool> OnValidation;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="TextMaskField"/> class using <see cref="LayoutStyle.Computed"/> positioning.
+		/// </summary>
+		protected TextMaskField ()
```

Comment:
I believe this needs to be public. We require all `View` classes to have a parameterless public constructor.

Like this

![image](https://user-images.githubusercontent.com/585482/115154584-795dd900-a030-11eb-887a-5aaa4d87ab2f.png)

Without this `AllVeiwsTester` crashes.

I think all unit test suites should have tests for each constructor, FWIW.